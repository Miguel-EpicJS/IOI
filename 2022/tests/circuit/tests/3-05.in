385045b28fbc1903eca997d513040235e0229e01
1000 1000 5
-1 0 1 1 2 0 0 6 6 8 8 8 9 11 7 11 14 16 16 7 14 7 20 19 7 20 20 25 17 18 12 27 29 32 11 29 17 8 37 7 7 18 31 42 40 40 29 41 14 46 48 40 32 35 18 41 54 32 43 32 57 57 32 37 45 64 56 47 54 62 41 50 34 47 48 37 36 63 25 12 45 65 46 33 37 49 25 86 86 65 64 84 86 64 25 85 41 51 41 71 87 92 75 36 29 93 34 34 62 90 51 51 88 40 7 104 90 89 105 66 110 22 117 65 65 121 116 97 86 108 70 88 110 50 112 122 57 97 65 88 125 87 108 114 6 88 138 27 101 138 114 125 114 104 144 124 92 80 57 125 78 88 136 82 121 67 114 105 163 151 36 144 171 122 117 171 115 171 81 14 59 121 172 81 155 178 165 117 132 139 71 64 86 6 183 119 120 114 51 166 166 172 139 124 138 114 166 206 206 125 110 77 207 155 114 153 108 122 128 195 112 203 188 199 221 125 141 51 138 88 186 98 89 146 184 226 206 138 134 106 169 239 236 214 138 169 154 51 211 216 166 236 191 231 156 96 121 151 226 183 148 248 215 236 187 243 214 200 227 259 255 242 232 136 251 164 249 89 273 276 208 14 176 246 246 157 280 259 237 172 169 276 45 271 232 208 166 267 190 265 104 142 296 170 223 278 194 259 171 217 296 206 198 266 229 288 228 266 316 145 242 305 215 296 317 56 296 319 300 231 239 272 230 116 288 123 310 300 250 337 174 22 253 292 186 336 290 243 253 295 250 231 216 104 347 116 243 225 212 188 329 270 74 325 174 328 17 288 326 284 18 279 77 363 370 348 353 339 343 300 138 277 292 325 109 224 220 312 314 379 199 255 338 305 381 323 257 306 193 278 379 238 284 326 244 333 286 333 407 259 249 153 229 248 218 408 314 229 403 310 258 395 379 418 422 350 333 320 418 336 314 403 423 386 137 355 245 386 400 363 403 404 326 440 296 296 358 292 427 197 428 444 278 336 320 337 332 335 382 403 438 209 414 292 283 231 336 443 389 366 341 468 54 412 336 459 282 157 410 466 469 431 463 405 481 411 479 458 185 444 445 290 440 117 445 155 277 477 472 445 498 417 469 384 336 489 504 410 232 472 461 311 352 201 466 493 495 499 492 292 513 518 502 499 401 414 444 494 435 460 502 499 530 499 533 313 477 528 526 296 492 466 485 518 39 348 535 519 422 546 296 514 523 493 491 550 554 498 459 295 458 551 403 178 472 547 431 492 514 440 372 561 509 531 407 379 514 540 565 522 548 472 561 533 302 500 535 516 508 550 533 487 567 496 559 566 580 543 564 576 598 590 487 576 194 492 295 524 574 520 342 538 586 394 427 422 382 422 454 432 498 70 249 463 588 585 227 310 555 533 548 558 429 629 625 598 511 437 578 628 569 583 622 453 459 643 400 646 641 415 584 608 473 499 581 618 616 626 623 296 653 567 403 640 553 517 653 666 484 564 662 526 507 665 569 548 590 663 552 651 657 534 575 648 481 629 642 343 309 370 590 590 590 340 689 665 598 476 635 694 690 652 663 671 668 700 589 679 479 627 535 708 670 701 575 702 694 696 418 686 584 600 417 675 687 502 703 708 702 615 540 623 509 721 706 658 709 535 694 709 502 644 529 718 725 604 676 533 666 614 597 741 730 499 753 700 724 498 747 733 746 758 730 758 735 689 758 716 765 266 749 765 736 754 706 761 769 753 761 709 753 582 747 627 783 743 545 765 762 705 577 758 718 791 783 747 789 766 794 711 775 768 801 741 687 738 778 798 767 807 748 801 502 799 360 807 793 783 791 787 768 800 656 777 802 770 809 699 695 723 738 665 732 403 791 818 785 794 814 767 782 776 814 805 763 840 795 841 644 786 838 692 835 817 850 803 772 379 853 844 834 692 856 805 831 863 837 780 811 817 843 834 857 862 870 864 846 849 822 857 798 418 879 754 883 821 878 857 827 794 755 853 805 891 855 881 872 877 872 832 886 823 692 754 903 895 743 902 902 812 902 903 910 902 875 864 896 846 887 905 912 913 908 902 912 854 902 902 848 907 915 888 931 931 911 917 758 927 879 936 903 939 936 921 918 941 936 882 892 939 926 936 919 946 936 927 761 923 942 884 904 949 961 962 939 954 964 934 940 954 936 918 971 953 973 919 965 860 942 915 953 980 977 959 978 961 977 950 985 987 985 970 923 985 987 943 994 979 906 956 3 4 5 5 2 2 5 14 10 14 10 16 21 15 20 16 20 17 20 18 24 25 23 26 36 25 31 36 33 25 15 30 38 53 59 61 52 62 70 55 56 57 75 49 28 55 69 76 13 54 63 83 54 40 81 44 58 72 61 85 73 97 65 77 100 90 37 93 87 103 116 80 59 87 111 109 79 70 103 87 121 82 138 134 94 92 139 49 93 115 99 60 126 86 59 118 153 130 93 89 130 102 151 26 84 38 165 140 129 143 162 164 126 147 168 129 181 133 169 149 160 190 139 100 166 107 151 163 158 171 183 186 137 96 200 87 82 95 200 179 128 159 172 152 203 172 210 154 175 210 91 203 173 202 181 127 196 182 220 166 245 235 128 155 68 234 259 240 218 264 121 135 217 231 247 264 241 250 179 254 82 232 263 219 175 171 253 139 280 157 112 236 259 230 281 250 205 166 253 204 203 252 226 211 290 261 291 203 302 289 96 192 284 315 184 311 19 148 243 306 321 287 161 221 260 308 320 324 303 246 307 151 300 323 269 315 337 329 218 335 197 150 342 308 222 233 163 205 280 326 345 155 311 259 71 293 256 171 271 167 309 367 336 368 360 299 314 303 329 238 360 342 209 251 374 365 351 305 377 294 370 327 364 359 369 274 379 314 82 379 372 403 221 263 372 341 397 113 238 373 404 344 399 398 181 392 318 215 301 282 262 363 306 291 354 325 402 344 358 380 426 391 419 400 408 433 435 274 425 413 434 252 380 389 360 297 364 421 349 385 132 336 376 388 275 441 150 438 373 382 189 423 422 375 420 427 411 285 234 458 428 280 410 444 444 439 433 388 382 275 396 185 368 389 430 289 470 468 362 265 177 446 447 357 447 434 416 440 400 356 304 383 455 402 482 370 227 218 458 215 404 452 334 467 476 453 480 393 495 400 458 475 361 501 330 353 475 378 399 478 464 486 483 476 269 512 453 131 403 375 427 448 451 422 369 457 516 409 387 456 511 514 517 521 527 179 479 502 445 449 384 372 505 390 539 466 441 466 532 484 466 484 331 436 371 422 495 424 471 460 463 172 517 381 437 526 548 470 534 554 503 542 288 529 562 353 206 488 469 558 536 180 543 128 539 406 512 440 538 573 298 302 510 535 505 525 550 452 322 548 544 516 552 573 490 293 524 475 498 497 548 546 593 503 596 564 599 495 594 587 541 462 586 445 346 461 526 527 560 603 556 599 546 597 601 465 611 612 572 592 560 450 605 619 615 626 640 588 541 515 615 537 629 632 517 607 645 621 517 563 608 571 593 479 645 605 507 643 652 642 568 474 647 659 521 643 571 656 613 533 607 654 639 213 579 620 584 625 630 602 600 506 595 674 672 650 658 622 678 557 663 606 549 677 662 666 649 629 593 624 638 555 484 631 637 676 668 680 669 650 696 664 617 697 673 702 594 676 597 674 674 667 634 631 487 688 627 447 694 705 693 649 710 633 665 720 678 505 720 714 717 660 576 705 661 691 684 694 710 691 722 671 655 721 570 734 712 731 697 608 636 442 699 724 609 667 727 720 661 714 713 489 627 688 689 659 704 610 743 740 728 682 733 687 744 738 733 725 698 711 729 700 740 739 737 749 683 591 685 748 719 740 582 715 753 745 765 751 268 681 770 762 689 753 779 765 759 786 698 778 754 775 794 752 781 784 800 781 648 798 765 756 794 753 743 815 705 797 801 806 750 804 755 806 707 810 740 806 790 814 755 806 760 808 812 553 827 762 820 757 807 764 670 806 812 803 769 839 820 841 742 837 725 819 842 792 833 773 802 850 726 816 852 820 771 851 827 796 860 836 769 860 846 779 858 864 820 842 735 865 829 828 850 821 821 826 761 776 824 869 830 619 838 865 870 840 864 871 867 847 749 868 866 873 788 846 876 879 861 885 813 896 893 878 871 791 774 873 889 881 849 825 879 823 894 761 891 880 904 905 845 859 838 900 903 885 889 893 920 904 873 874 916 898 923 892 909 918 544 914 923 922 894 932 928 885 783 927 876 931 932 872 920 925 937 901 848 902 909 780 894 944 930 940 929 922 909 940 946 942 891 898 936 902 942 897 947 958 954 960 935 953 857 938 899 965 966 951 924 950 972 933 963 960 957 976 966 978 952 961 964 978 890 960 977 975 968 974 946 977 984 967 970 948 958 988 987 989 983 941 955 981 980 961 982 976 985 988 804 994 989 969 985 992 967 994 986 945 987 997 982 994 990 993 998 996 995 993 999 991
1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
1774 1919
1426 1788
1377 1852
1218 1584
1286 1691
