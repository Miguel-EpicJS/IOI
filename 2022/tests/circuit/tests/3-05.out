4faa78c22ee35083e49b65601698a1bd98a87a0f
OK
814713862
179971074
530108958
172714996
724155456
